airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py

```diff
@@ -0,0 +1,70 @@
+import sys
+from typing import Any, Callable, Optional
+
+import dagger
+from asyncclick import Context, get_current_context
+from dagger.api.gen import Client, Container
+from pydantic import BaseModel, Field, PrivateAttr
+
+from ..singleton import Singleton
+
+
+class ClickPipelineContext(BaseModel, Singleton):
+    dockerd_service: Optional[Container] = Field(default=None)
+    _dagger_client: Optional[Client] = PrivateAttr(default=None)
+    _click_context: Callable[[], Context] = PrivateAttr(default_factory=lambda: get_current_context)
+
+    @property
+    def params(self):
```

Comment:
I'd love a docstring to understand the purpose of this property.
I believe its content is static, should we add the params attribute it at init time?

Reply:
I think we add to the params when we add arguments to click commands? 

I've been pulling params from this property in the way that I used to access the other type of context, i.e. 

If ctx is click.Context: `ctx.obj.get("key")`
if ctx is ClickPipelineContext: `ctx.params.get("key")`

and I guess the idea is that these `params` contain both the click context object keys and the click command arguments

Would love confirmation :)

Reply:
That is the idea! Though. It doesnt have to be. But the idea was to always be merging the idea of parent arguments/options with current argument/options so the way you reference "branch_name" or "connector_name" is the same no matter what level of the CLI command hierarchy youre at

Reply:
It's not a blocker for me, would just appreciate a docstring 😄